diambra/arena/env_settings.py

```diff
@@ -1,200 +1,389 @@
-            check_val_in_list("action_space[{}]".format(jdx), self.action_space[jdx], ["discrete", "multi_discrete"])
+        check_num_in_range("n_players", self.n_players, [2, 2])
+        char_list = list(self.env_info.characters_info.char_list)
+        char_list.append("Random")
+        for idx in range(2):
+            check_val_in_list("action_space[{}]".format(idx), self.action_space[idx], ["discrete", "multi_discrete"])
+            check_val_in_list("role[{}]".format(idx), self.role[idx], ["P1", "P2", "Random"])
+            for jdx in range(3):
+                check_val_in_list("characters[{}][{}]".format(idx, jdx), self.characters[idx][jdx], char_list)
+            check_num_in_range("outfits[{}]".format(idx), self.outfits[idx], self.games_dict[self.game_id]["outfits"])
+            check_val_in_list("super_art[{}]".format(idx), self.super_art[idx], ["Random", 1, 2, 3])
+            check_val_in_list("fighting_style[{}]".format(idx), self.fighting_style[idx], ["Random", 1, 2, 3])
+            for jdx in range(3):
+                check_val_in_list("ultimate_style[{}][{}]".format(idx, jdx), self.ultimate_style[idx][jdx], ["Random", 1, 2])
+
+    def _process_random_values(self):
+        super()._process_random_values()
+
+        characters_tmp = [[],[]]
+        for idx in range(2):
+            sampled_characters = self._sample_characters()
+
+            for jdx in range(3):
+                if self.characters[idx][jdx] == "Random":
+                    characters_tmp[idx].append(sampled_characters[jdx])
+                else:
+                    characters_tmp[idx].append(self.characters[idx][jdx])
+
+        self.characters = (tuple(characters_tmp[0]), tuple(characters_tmp[1]))
+
+        if self.role[0] == "Random":
+            if self.role[1] == "Random":
+                idx = random.choice([1, 2])
+                self.role = ("P{}".format(idx), "P{}".format((idx % 2) + 1))
+            else:
+                self.role = ("P1" if self.role[1] == "P2" else "P2", self.role[1])
+        else:
+            if self.role[1] == "Random":
+                self.role = (self.role[0], "P1" if self.role[0] == "P2" else "P2")
+        self.super_art = tuple([random.choice(list(range(1, 4))) if self.super_art[idx] == "Random" else self.super_art[idx] for idx in range(2)])
+        self.fighting_style = tuple([random.choice(list(range(1, 4))) if self.fighting_style[idx] == "Random" else self.fighting_style[idx] for idx in range(2)])
+        self.ultimate_style = tuple([[random.choice(list(range(1, 3))) if self.ultimate_style[idx][jdx] == "Random" else self.ultimate_style[idx][jdx] for jdx in range(3)] for idx in range(2)])
+
+    def _get_action_spaces(self):
+        action_spaces = [model.EnvSettings.ActionSpace.ACTION_SPACE_DISCRETE if self.action_space[idx] == "discrete" else \
+                         model.EnvSettings.ActionSpace.ACTION_SPACE_MULTI_DISCRETE for idx in range(2)]
```

Comment:
Same again

Reply:
Updated, as said before, I kind of did that on purpose to make sure these limits will be clear when willing to move to more agents. But I agree is not clean. Thanks